File: apps/backend/src/rhesis/backend/app/schemas/type_lookup.py
from typing import Optional

from pydantic import UUID4

from rhesis.backend.app.schemas import Base


# Category schemas
class TypeLookupBase(Base):
    type_name: str
    type_value: str
    description: Optional[str] = None
    organization_id: Optional[UUID4] = None
    user_id: Optional[UUID4] = None


class TypeLookupCreate(TypeLookupBase):
    pass


class TypeLookupUpdate(TypeLookupBase):
    type_name: Optional[str] = None
    type_value: Optional[str] = None


class TypeLookup(TypeLookupBase):
    pass

File: apps/frontend/src/utils/api-client/interfaces/type-lookup.ts
import { UUID } from 'crypto';

// Base interface for type lookups
export interface TypeLookupBase {
  type_name: string;
  type_value: string;
  description?: string;
}

export type TypeLookupCreate = TypeLookupBase;

export type TypeLookupUpdate = Partial<TypeLookupBase>;

export interface TypeLookup extends TypeLookupBase {
  id: UUID;
}

export interface TypeLookupsQueryParams {
  skip?: number;
  limit?: number;
  sort_by?: string;
  sort_order?: 'asc' | 'desc';
  $filter?: string;
}
